lib/blob.js

```diff
@@ -0,0 +1,377 @@
+  for (var key in options) {
+    if (options.hasOwnProperty(key) && options[key] !== undefined) {
+      this.options[key] = options[key];
+    }
+  }
+
+  // Validate options
+  assert(this.options.accountId, "`options.accountId` must be given");
+
+  // Construct hostname
+  this.hostname = this.options.accountId + '.blob.core.windows.net';
+
+  // Compute `timeout` for client-side timeout (in ms), and `timeoutInSeconds`
+  // for server-side timeout in seconds.
+  this.timeout = this.options.timeout + this.options.clientTimeoutDelay;
+  this.timeoutInSeconds = Math.floor(this.options.timeout / 1000);
+
+  // Define `this.authorize`
+  if (this.options.accessKey) {
+    // If set authorize to use shared key signatures
+    this.authorize = authorizeWithSharedKey;
+    // Decode accessKey
+    this._accessKey = new Buffer(this.options.accessKey, 'base64');
+  } else {
+    throw new Error("Either options.accessKey, options.sas as function or " +
+      "options.sas as string must be given!");
+  }
+};
+
+// Export Blob
+module.exports = Blob;
+
+/**
+ * Construct authorized request options by adding signature or
+ * shared-access-signature, return promise for the request options.
+ *
+ * @protected
+ * @method authorize
+ * @param {string} method - HTTP verb in upper case, e.g. `GET`.
+ * @param {string} path - Path on blob resource for storage account.
+ * @param {object} query - Query-string parameters.
+ * @param {object} header - Mapping from header key in lowercase to value.
+ * @returns {Promise} A promise for an options object compatible with
+ * `https.request`.
+ */
+Blob.prototype.authorize = function(method, path, query, headers) {
+  throw new Error("authorize is not implemented, must be defined!");
+};
+
+/**
+ * Make a signed request to `path` using `method` in upper-case and all `query`
+ * parameters and `headers` keys in lower-case. The request will carry `data`
+ * as payload and will be retried using the configured retry policy,
+ *
+ * @private
+ * @method request
+ * @param {string} method - HTTP verb in upper case, e.g. `GET`.
+ * @param {string} path - Path on blob resource for storage account.
+ * @param {object} query - Query-string parameters.
+ * @param {object} header - Mapping from header key in lowercase to value.
+ * @return {Promise} A promise for HTTPS response with `payload` property as
+ * string containing the response payload.
+ */
+Blob.prototype.request = function request(method, path, query, headers, data) {
+  // Set timeout, if not provided
+  if (query.timeout === undefined) {
+    query.timeout = this.timeoutInSeconds;
+  }
+
+  // Set date, version and client-request-id headers
+  headers['x-ms-date']              = new Date().toUTCString();
+  headers['x-ms-version']           = this.options.version;
+  headers['x-ms-client-request-id'] = this.options.clientId;
+
+  // Construct authorized request options with shared key signature or
+  // shared-access-signature.
+  var self = this;
```

Comment:
Too bad we still support 0.12 for `azure-entities`.  @jonasfj what do you think about making Azure-Entities and this repository require node >= v4?